generators/client-2/templates/src/main/webapp/app/shared/directive/jhi-translate.ts

```diff
@@ -0,0 +1,28 @@
+import { Directive, ElementRef, HostListener, Input, Renderer, OnInit, TemplateRef } from '@angular/core';
+
+
+@Directive({
+  selector: '[jhiTranslate]',
+  exportAs: 'jhiTranslate'
+})
+export class JhiTranslate {
+
+  private _jhiTranslate: string | TemplateRef<any>;
+ 
+  constructor(private el: ElementRef, private renderer: Renderer) { }
+
+  @Input()
+  set jhiTranslate(value: string | TemplateRef<any>) {
+      this._jhiTranslate = value;
+      this.setTranslatedValue();
+  }
+
+  setTranslatedValue() {
+    if (this._jhiTranslate ) {
```

Comment:
use service to translate key so that we can avoid the pipe and do `jhi-translate="home.title"` https://github.com/ocombe/ng2-translate#4-use-the-service-or-the-pipe